pytorch_lightning/trainer/logging.py

```diff
@@ -57,6 +58,23 @@ def process_dict_result(self, output, train=False):
 
         Separates loss from logging and progress bar metrics
         """
+        # --------------------
+        # WARN DEPRECATED KEYS
+        # --------------------
+        # TODO: 1.0.0 remove
+        if isinstance(output, dict):
+            for k, v in output.items():
+                if k in ['log', 'progress_bar']:
+                    m = f"""
+                    The {{{k}:dict keyword}} was deprecated in 0.9.1 and will be removed in 1.0.0
+                    Please use self.log(...) inside the lightningModule instead.
+ 
+                    # log on a step or aggregate epoch metric to the logger and/or progress bar
+                    # (inside LightningModule)
+                    self.log('train_loss', loss, on_step=True, on_epoch=True, prog_bar=True)
+                    """
```

Comment:
Please, do not use triple quote strings for warnings and logs since they include all the extra whitespace:

```
>>> import warnings
>>> m1 = (
...     "this is "
...     "a "
...     "test"
... )
>>> warnings.warn(m1)
<stdin>:1: UserWarning: this is a test
>>> m2 = """
...   this
...   is a
...   test
... """
>>> warnings.warn(m2)
<stdin>:1: UserWarning: 
  this
  is a
  test

```

Reply:
i disagree with this... 

these long strings are more readable to the user when the get this kind of warning.

Reply:
The problem is not that it is multiline, but having the whitespace at the beginning depends on the indentation level.

For that case in particular:
```
<stdin>:1: UserWarning: 
                    The {{{k}:dict keyword}} was deprecated in 0.9.1 and will be removed in 1.0.0
                    Please use self.log(...) inside the lightningModule instead.
 
                    # log on a step or aggregate epoch metric to the logger and/or progress bar
                    # (inside LightningModule)
                    self.log('train_loss', loss, on_step=True, on_epoch=True, prog_bar=True)

```

You would have to either remove the indentation:
```python
                    m = f"""
The {{{k}:dict keyword}} was deprecated in 0.9.1 and will be removed in 1.0.0
Please use self.log(...) inside the lightningModule instead.

# log on a step or aggregate epoch metric to the logger and/or progress bar
# (inside LightningModule)
self.log('train_loss', loss, on_step=True, on_epoch=True, prog_bar=True)
                    """
```
(pretty ugly if you ask me)

or split it and print it with more than one warnings/logs

Reply:
I agree with @carmocca, if you need an extra line, just use `\n`